src/events.js

```diff
@@ -20,7 +20,23 @@ module.exports = function(ctx) {
 
   var events = {};
   var currentModeName = Constants.modes.SIMPLE_SELECT;
-  var currentMode = ModeHandler(modes.simple_select(ctx), ctx);
+  var currentMode;
+
+  function initialize() {
+    // Add event listeners
+    ctx.map.on('mousemove', events.mousemove);
+
+    ctx.map.on('mousedown', events.mousedown);
+    ctx.map.on('mouseup', events.mouseup);
+
+    if (ctx.options.keybindings) {
+      ctx.container.addEventListener('keydown', events.keydown);
+      ctx.container.addEventListener('keyup', events.keyup);
+    }
+
+    // Start us in simple_select mode
+    currentMode = ModeHandler(modes.simple_select(ctx), ctx);
```

Comment:
The goal was to ensure that `ctx.map` is available in `simple_select`'s `start` method, so we can get its container and attach a listener to it. The way things were before, `simple_select` was first initialized before the map was ready.